### PR TITLE
s22-6pm-2 Yufei  backend API Add a Course to Personal Course Database 

### PR DESCRIPTION
**Pull request by Sean-syf**
Given a personal schedule id, enroll code, and quarter, add one data to the personal course database.

Todo:
Determine whether the enroll code is valid in the given quarter.
Waiting for UCSB course search API to finish.

Screen Shot 2022-05-28 at 4 01 54 PM

### CODE REVIEW

**Review by austinmacmath — DISMISSED** [review]
lgtm

File: src/main/java/edu/ucsb/cs156/courses/controllers/PersonalCoursesController.java [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+            @ApiParam("psID") @RequestParam long psId,
+            @ApiParam("Quarter (in yyyyq)") @RequestParam String quarter) {
+        CurrentUser currentUser = getCurrentUser();
+        log.info("currentUser={}", currentUser);
+
+        PersonalCourses personalcourse = new PersonalCourses();
+        personalcourse.setEnrollCd(enrollCd);
+        personalcourse.setQuarter(quarter);
+        personalcourse.setPsId(psId);        
+        //personalcourse.getPersonalSchedule().setDescription(description);;
```
Comment (pconrad) [inline src/main/java/edu/ucsb/cs156/courses/controllers/PersonalCoursesController.java:63]: Can we remove this commented out line of code?

**Review by pconrad — CHANGES_REQUESTED** [review]
LGTM except for one line of commented out code that we should probably remove.

I'll test on Heroku in the meantime.

**Comment by pconrad** [comment]
LGTM.  It would be good to add some validation, i.e. 
* the enrollCd should be numeric and no more than five digits
* the psId should be the id of an existing personal schedule in the personal schedule table
* the quarter should be a legal quarter in yyyyq format, i.e. yyyy should be between 1900 and 2099, and q should be 1, 2, 3 or 4

And if any of those are not true, the controller throws an IllegalArgumentException.

But that can be a separate issue that someone can work on.  (I'll add it as an issue.)

**Review by pconrad — APPROVED** [review]

**Pull request merged**